### DISCUSSION TITLE
x-data allows for multiple data can I do similar with :class

### DISCUSSION THREAD

**Discussion by StanleyCoding** [alpinejs/alpine#965] [1. Help]
Only been coding in Alpine (and Tailwind) for less than a week - already massively impressed by the potential. Pretty much every obstacle I have encountered so far I have been able to overcome but this one has me a little stuck. I have developed a workaround (see below) of having a child wrapper div inside but would be curious to know if there is a solution in case this scenario arises again.

Is it possible to have two classes on the same element (happens to be body tag in this case but could have been any element) where one can be updated as a property of an object triggered on a button using `changeTheme()` and the other is a class object whose key is a class name and the boolean is toggled elsewhere using button events, if any one of the two conditions is true then the text-xl class is applied.

Currently this all works as I want but keen to know if I can do what I achieve while doing it via the one :class.

**HTML**
```
<body class="bg-grey-dark" x-data="{...global.init(),...theme.init(),...theme.content()}" :class="{'text-xl': isSettings || isPost }">
<div :class="themeClass">
```

**Javascript**
```
var global = {
    init: function(){
        return {
            isSettings: false,
            isPost: false
        };
    }
};
const colors = ['color1','color2','color3','color4','color5','color6'];
const currentTheme = colors[Math.floor(Math.random() * colors.length)];
var theme = {
    init: function(){
        return {
            themes: colors,
            themeClass: currentTheme,
            changeTheme(className){
                // Update themeClass to the new theme value
                this.themeClass = this.themes.find(function(value){ return value === className; });
            }
        };
    },
    content: function(){
        return {
            //More code in here
        };
    }
};
```

**Comment by KevinBatdorf** [comment]
Does this work maybe?

```html
<div :class="{'text-xl': isSettings || isPost, [themeClass]: true }>
```

Also, having the component on the `<body>` tag might not be a good idea especially if you're doing a lot of expensive work as it will run on every single mutation. Better to scope your components small and dispatch events, etc

**Reply by StanleyCoding** [reply]
I appreciate the response and I'm still trying to get a handle on the scope (it appears to have some limitations I've seen discussed such as a  problem with nested scope) that's why I've put it all on the body at this stage. Still experimenting but I do realise its not ideal.

I have previously tried what you suggested. `themeClass` value initialises the :class as expected and it updates but only once then after that the `themeClass` part of :class stops updating even though the object property continues to update. When its separated like the workaround it can be changed infinitely as the :class keep updating in synch with the object property value changes.

**Reply by KevinBatdorf** [reply]
Another way to try would be to use a method:

```html
<div :class="classes">
```
```js
return {
  get classes() {
    return ''
  }
}
```

**Reply by StanleyCoding** [reply]
Again, thanks for the input. I tried something similar previously and I tried it again. Doesn't quite work,I'll need to get a better handle on scope as its appears to be directly related to my problem.
What I'm doing at the moment is very light work and the workaround is fine but wanted to get a handle on it in the event of doing something more complicated with heavier work.

**Reply by KevinBatdorf** [reply]
I would probably just go with what works then. Another option is an inline ternary, so like this:

```html
<div :class="somethingToCheck ? 'one-class' : 'two-class'">
```

I think what breaks in the first suggestion I gave might be a bug actually. It seems to preserve the class regardless of whether it updates.

**Reply by StanleyCoding** [reply]
Many thanks. As I said I'm very new to Alpine and to be honest I don't think it would be right choice for what I'm doing with my experimenting but it's serving as a good way to get a degree of understanding of things.
